engine/classes/Elgg/EntityIconService.php

```diff
@@ -185,81 +185,132 @@ public function saveIconFromElggFile(ElggEntity $entity, ElggFile $file, $type =
 	 */
 	public function saveIcon(ElggEntity $entity, ElggFile $file, $type = 'icon', array $coords = []) {
 
+		$type = (string) $type;
+		if (!strlen($type)) {
```

Comment:
`strlen()` on a non-string value will generate an error. Just check if `empty`

Reply:
thats why i cast it to a string 1 line above

Reply:
`(string) false` is `0`; `strlen('0')` is `1`